File: elixir/string-matching-in-an-array.ex
defmodule Solution do
  @spec string_matching(words :: [String.t]) :: [String.t]
  def string_matching(words), do:
      words
      |> Enum.filter(fn word -> Enum.any?(words, &(&1 != word and String.contains?(&1, word))) end)
end

File: javascript/string-matching-in-an-array.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var stringMatching = function(words) {
    words.sort((a, b) => a.length - b.length);
    const substrings = [];
    for (let i = 0; i < words.length; i++) {
        for (let j = i + 1; j < words.length; j++) {
            if (words[j].includes(words[i])) {
                substrings.push(words[i]);
                break;
            }
        }
    }
    return substrings;
};
